controller/setting_controller.go

```diff
@@ -896,6 +900,32 @@ func (sc *SettingController) syncDefaultLonghornStaticStorageClass() error {
 	return err
 }
 
+func (sc *SettingController) syncOrphanResourceAutoDeletion() error {
+	setting, err := sc.ds.GetSettingOrphanResourceAutoDeletion()
+	if err != nil {
+		return err
+	}
+	var replicaDataCleanupValue string
+	if setting[types.OrphanResourceTypeReplicaData] {
+		replicaDataCleanupValue = "true"
+	} else {
+		replicaDataCleanupValue = "false"
+	}
+
+	deprecatedSetting, err := sc.ds.GetSetting(types.SettingNameOrphanAutoDeletion)
+	if err != nil {
+		return err
+	}
+	if replicaDataCleanupValue != deprecatedSetting.Value {
+		deprecatedSetting.Value = replicaDataCleanupValue
+		if _, err := sc.ds.UpdateSetting(deprecatedSetting); err != nil {
+			return err
+		}
+	}
```

Comment:
Do we need to update the deprecatedSetting? Will the deprecated setting value be reset the next time user apply the Longhorn deployment yaml?

Reply:
1. syncing the new config to deprecated setting would eliminate the user confusion about whether the data store auto cleanup is enabled or not.
2. the new setting will first be synced from old setting in LH manager's upgrade stage, then keep syncing the new setting back to the old one after the setting controller launched.